python/lsst/daf/butler/_butler.py

```diff
@@ -2409,7 +2415,11 @@ def validateConfiguration(
             if "instrument" in datasetType.dimensions:
                 for instrument in instruments:
                     datasetRef = DatasetRef(
-                        datasetType, {"instrument": instrument}, conform=False  # type: ignore
+                        datasetType,
+                        {"instrument": instrument},  # type: ignore
+                        conform=False,
+                        run="validate",
+                        id=uuid.uuid4(),
```

Comment:
Do you need this, I thought IDs are now generated automatically if run is provided?

Reply:
Correct. After my rewrite I failed to spot that I had added this in a previous commit.